### ISSUE TITLE
[Taxes] Tax rate deleted on Old Dot is not removed from New Dot

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#38559]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.54-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
Precondition:
- User is an admin of Collect workspace.

1. Go to staging.new.expensify.com.
2. Go to Profile > Workspaces > Collect workspace.
3. Go to Taxes.
4. Add two tax rates.
5. Go to Old Dot.
6. Delete one of the tax rates.
7. Refresh the page on New Dot.

## Expected Result:
The tax rate deleted in Step 6 will be removed from New Dot.

## Actual Result:
The tax rate deleted in Step 6 remains in New Dot. When adding the deleted tax rate as Workspace currency default, it shows error "Invalid tax for default currency".

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/829259e0-8dff-4ed6-b6f8-4ba2866f6709

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ae82030c838a47dc</li>
        <li>Upwork Job ID: 1770021818639007744</li>
        <li>Last Price Increase: 2024-03-19</li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
@dylanexpensify I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.


We think that this bug might be related to the #vip-vsb.

**Comment by neil-marcellini** [comment]
@dylanexpensify @izarutskaya I don't think this is a blocker because the only reason it's not reproducible on prod is that you can't add tax rates from NewDot on prod yet. I am able to reproduce the issue if I simply add the tax rates on OldDot vs NewDot.

It will likely require backend changes to push an update to properly delete the tax rate on NewDot, so it should be internal.

https://github.com/Expensify/App/assets/26260477/39bf59d2-0350-4a37-a9e7-fa588a35b50c

**Comment by neil-marcellini** [comment]
@dylanexpensify would you please help find someone to take this over who is working on tax for collect. I would like to stay focused on other stuff. If you have trouble finding someone I can work on it.

**Comment by dylanexpensify** [comment]
@neil-marcellini can do!!

**Comment by dylanexpensify** [comment]
@neil-marcellini asking Carlos [here](https://expensify.slack.com/archives/C036QM0SLJK/p1710841262557019?thread_ts=1710800286.055929&cid=C036QM0SLJK)!

**Comment by mountiny** [comment]
I have dug into this one and it seems like its the same issue I have been seeing for a while around but not sure now ahts the latest.

When the tax is deleted, we send a correct onyx update something like this:

```
{"key":"policyTaxRates_2952FD9712138996","onyxMethod":"merge","value":{"taxes":{"id_TAX_OPTION_3":null}}}
```

which should remove the `id_TAX_OPTION_3` key from the taxes. However, as its nested, onyx does not process the key removal correctly it seems.

There are some issues going on already to address similar problems https://github.com/Expensify/App/issues/37560

This is not only related to the tax page so I am going to put this issue on hold on the linked one and keep an eye on it

**Comment by mountiny** [comment]
ok this might be different issue

**Comment by mountiny** [comment]
😂 yeah I just noticed comparing this to Tags

**Comment by mountiny** [comment]
So once this is done we should be good https://github.com/Expensify/App/issues/36171

**Comment by dylanexpensify** [comment]
held

**Comment by dylanexpensify** [comment]
same

**Comment by mountiny** [comment]
I believe this should be fixed now, need to retest

**Comment by trjExpensify** [comment]
@dylanexpensify did you retest this one? Can we close it?

**Comment by mountiny** [comment]
I am going to retest locally, was able to reproduce 3 days ago, but I am not sure if that was my local issue

**Comment by dylanexpensify** [comment]
Can still reproduce

**Comment by mountiny** [comment]
https://github.com/Expensify/App/assets/36083550/58b6e4a7-e70a-46f9-9a7d-d694755f269f


I could not reproduce now locally

**Comment by trjExpensify** [comment]
Haha, so what's Dylan doing differently? 🤔

**Comment by mountiny** [comment]
I am testing this locally, havent tested in staging

**Comment by trjExpensify** [comment]
Got it, but steps wise are we sure everyone is doing the same? :)

**Comment by mountiny** [comment]
I have also tested staging and I could reproduce the issue, the tax rows were not deleted or added dynamically. I also could not really delete categories in staging optimistically, but it works locally

**Comment by mountiny** [comment]
Thanks, I think this is just some flakiness with the onyx updates, but it should be working, I have tested locally and it works, @hoangzinh confirmed staging working too. I am closing this for now
